changes/807-timonbimon.rst

```diff
@@ -0,0 +1 @@
+add support for Type[T] type hints when arbitrary_types_allowd is True
```

Comment:
```suggestion
add support for Type[T] type hints when arbitrary_types_allowd is True
```
```suggestion
add support for ``Type[T]`` type hints
```
I don't think support for `Type[...]` should require `arbitrary_types_allowd`

Reply:
sure I'll add it out of the box (never really understood why arbitrary_types_allowed is not True by default).

we are then allowing for arbitrary user-defined classes though (which seemed analogue to arbitrary_types_allowed to me)

Reply:
Maybe we could set `arbitrary_types_allowed` to `True` before v1, the reason it's false was to avoid breaking chances when it was introduced. That's a different discussion, please create an issue if you feel strongly about it.

I understand these two things are related but they're not the same:
* `foo: Type[Foobar]` says "`foo` should be the thing `Foobar` or a subclass of that thing". I guess we should check that `issubclass` works with `Foobar`, eg. `Foobar` should inherit from `type`.
* `foo: Foobar` with `arbitrary_types_allowed=True` say, `foo` must be an instance of `Foobar`, but don't throw an error if `Foobar` doesn't implement `__get_validators__`. In other works the check is just `isinstance()` and nothing more, which differs from other types (custom or otherwise) in pydantic.

`arbitrary_types_allowed` exists to warn people that they haven't implemented `__get_validators__`, there's no equivalent with `Type[...]` so I don't think another config flag is required.

Reply:
alright makes sense! I'll open an issue for the arbitrary_types_allowed for v1 :)